src/storage/txn/actions/get_old_value.rs

```diff
@@ -9,39 +9,14 @@ use txn_types::{Key, OldValue, Write};
 /// it must be read in the caller and be passed in for optimization
 pub fn get_old_value<S: Snapshot>(
     txn: &mut MvccTxn<S>,
-    key: &Key,
+    _key: &Key,
     prev_write: Option<Write>,
 ) -> MvccResult<OldValue> {
-    // Precondition:
-    debug_assert!(if let Some(w) = &prev_write {
-        let cursor = txn.reader.write_cursor.as_ref().unwrap();
-        let key_under_cursor =
-            Key::from_encoded(cursor.key(&mut txn.reader.statistics.write).to_vec())
-                .truncate_ts()
-                .unwrap();
-        let write_under_cursor =
-            txn_types::WriteRef::parse(cursor.value(&mut txn.reader.statistics.write))
-                .unwrap()
-                .to_owned();
-        key.clone() == key_under_cursor && w.clone() == write_under_cursor
-    } else {
-        true
-    });
     match prev_write {
-        Some(w) if !w.may_have_old_value() => {
```

Comment:
So how do we handle the case now? it doesn't affect correctness?

Reply:
cdc worker will try to get old value from rocksdb when `prev_write` is `Rollback` or `Lock` type

Reply:
No much effect because those cases should be rare.